### ISSUE TITLE
[crash] if set mesh on obj recieves poly indices list as [()]

### ISSUE THREAD

**Issue by og76** [JacquesLucke/animation_nodes#349]
[meshData_check_issue.blend](https://drive.google.com/file/d/0BzzAoBwXO8dsSE5aTEJfTk5RSjQ/view?usp=sharing)
connect as indicated
we need a check for this

I know is coming from expression/script and "should not..", but it's one thing to halt execution of An or other errors, another to crash .
And I know you prevent it at the level of create poly indices etc, but maybe should be in **set mesh on obj** along with the other checks.

basically crash if poli indices list contains poly like ()

**Comment by og76** [comment]
[second file](https://drive.google.com/file/d/0BzzAoBwXO8dseTkzSzRLSjJkLTQ/view?usp=sharing)

here I define the poly indices (not list) as []
if u try to connect (that would result in create list..) crash already

so [] or (), all the same .., and even if it goes thru create list ..

**Comment by JacquesLucke** [comment]
yeah I know that the mesh creation creation funktion really depends on the fact that the input data is correct. I can implement a test method to check if everything is correct, but I guess this can become really slow for larger meshes.

**Comment by og76** [comment]
or maybe at the socket level ? to not accept [()] or [[]], respectively () [] ?
or are there blender check functions to use? besides ensure lookup or so..

Just a wild guess.I have no idea how it should be
+ I guess it happens only with script or expression afaict

**Comment by JacquesLucke** [comment]
yes if you only use the normal nodes without custom scripts you will not be able to crash Blender this way (at least my plan was that you cannot do so).

I'm  right now implementing this check. I will make it an option in the advanced settings.
